src/unix/mmap.c

```diff
@@ -492,15 +507,25 @@ static boolean vmap_update_protections(heap h, rangemap pvmap, range q, u32 newf
     thread_log(current, "%s: validate %R", __func__, q);
     if (!validate_user_memory(pointer_from_u64(q.start), range_span(q), false) ||
         rangemap_range_find_gaps(pvmap, q, stack_closure(vmap_update_protections_gap))) {
-        return false;
+        return -ENOMEM;
+    }
+
+    rmnode_handler v_handler = stack_closure(vmap_update_protections_validate,
+        newflags, false);
+    if (!rangemap_range_lookup(pvmap, q, v_handler)) {
+        return -ENOMEM;
+    } else {
+        closure_ref(vmap_update_protections_validate, v);
```

Comment:
Why not just enclose a pointer to a local boolean rather than do this ref / cast to access prot_violation?

Reply:
Indeed I made it more complicated than it needs to be. Now I replaced this with a local boolean and let go of the closure_cast() thing.